### DISCUSSION TITLE
Firefly III - docker container with custom network - How to manually change default webui port?

### DISCUSSION THREAD

**Discussion by frogweh** [firefly-iii/firefly-iii#8230] [General]
Just as the title says, I use a custom network with my docker containers so each container has it's own IP scheme. I have been able to find ways to manually change services inside all of my other docker containers to default to port 80 so when I do my DNS rewrites I can just type in the address and don't have to worry about specifying a port. I would like to do this for Firefly III (as well as the importer) but I'm not really sure where to go to set it. Does anyone know this service better, and can point me in the right direction? I can't find any information on their website, or github. I'm not familiar with this service or how it runs so if all I need to do is change a config file somewhere I can absolutely do that. If this isn't possible, then I guess I'll do a feature request next. Thank you!

**Comment by JC5** [comment]
You can do `-p 80:8080` when you run the container, or change your docker compose file. There is no need to change things inside of Docker containers!

**Reply by frogweh** [reply]
I'm running the docker container inside of unraid, with a custom network. I use my own IP scheme, and when using your own IP scheme docker ignores port mappings. I have already confirmed this on several different docker forums, so I absolutely have to change the default port of the service inside the container. The port mapping doesn't work with a custom network with a set IP.

**Reply by JC5** [reply]
Welp, that's not something I'm going to make customizable.

**Reply by frogweh** [reply]
So I wasn't really asking for it to be customizable for everyone, unless it's hard coded somewhere. I definitely don't want you to have to re-write anything, because that's a pain in your ass. I'm more looking at if there's a config file for the web service in side the docker container I can go change that config file, or if it's running apache I can just set apache to default to the port I want or something. I'm just trying to come up with a solution for my setup that doesn't require anyone to do anything more than point me in the right direction. If that's not a viable solution at all, then I guess I'll have to figure something else out on my end.

**Reply by JC5** [reply]
You can expose another port and edit the apache config file :)

**Reply by frogweh** [reply]
Now we're cooking with gasoline! I'm not super familiar with apache, is the config file in a default location I can just google or is it in a custom location?

**Reply by JC5** [reply]
`/etc/apache2/sites-available/000-default.conf` and `/etc/apache2/ports.conf`, they both have references to port 8080.

**Reply by frogweh** [reply]
Thank you kindly, I'll take a look at that and see if I can get it working. I'll update this discussion with my findings and possible instructions in case anyone else would like to use this method. I really appreciate your help!

**Reply by frogweh** [reply]
Okay so what I had to do to get this to work and mount the /etc/apache2 folder to my host box is as follows. I found an empty folder that I could mount to my host system already that already existed inside the docker container (I used /mnt/ as it was empty and easily accessible). I edited the docker configs in Unraid and added a path to mount /mnt/ from inside the containers to a folder on my host machine (e.x. /mnt/user/appdata/firefly/apache2 OR /mnt/user/appdata/fidi/apache2). Once I had them mounted I went inside the console for the containers and I copied the entire contents of /etc/apache2/ to /mnt/

`cp /etc/apache2/* -R /mnt/`

I then edited the ports.conf file and the sites-available/000-default.conf, by accessing my Unraid terminal and running the following comands

`cd /mnt/user/appdata/<folder>/apache2`
`sed -i 's/8080/80/g' ports.conf`
`sed -i 's/8080/80/g' sites-available/000-default.conf`

I then reconfigured the docker containers to mount /etc/apache2/ from inside the container to my new apache2 folder (/mnt/user/appdata/firefly/apache2) so now the config files should persist between updates.

The reason I had to copy the contennts of the /etc/apache2 directories is when I just tried to mount those folders to my host system, the docker container kept crashing. After looking through the logs it looks like it's because it wasn't putting the files in the folder before mounting. It was attempting to run config files that didn't exist. So I had to get them OUT of the container in order to put them on my host machine. It's a little tedious but definitely not something you need advanced knowledge for.

I hope this helps anyone else who was looking to change the ports inside the docker containers!
